### PR TITLE
Converted boost tests into gtests #647

### PR DESCRIPTION
**Pull request by lodle**

### CODE REVIEW

File: src/shared/uicore/code/UICoreEntry.cpp [hunk 1]
```diff
@@ -94,6 +96,16 @@ class UICore : public UICoreI
 
 		m_bExitCodeSet = false;
 		m_iExitCode = 0;
+
+#ifdef DEBUG
+#ifdef WITH_GTEST
+#ifdef WIN32
```
Comment (karolherbst) [inline src/shared/uicore/code/UICoreEntry.cpp:102]: why not in one line?
#if defined(DEBUG) && defined(WITH_GTEST) && defined(WIN32)

File: src/shared/unittests/code/util/MD5_test.cpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#define BOOST_TEST_MODULE MD5_test
```
Comment (karolherbst) [inline src/shared/unittests/code/util/MD5_test.cpp]: can be removed

File: src/shared/unittests/code/util/gcBuff_test.cpp [hunk 3]
```diff
@@ -0,0 +1,62 @@
+#define BOOST_TEST_MODULE gcBuff_test
```
Comment (karolherbst) [inline src/shared/unittests/code/util/gcBuff_test.cpp]: can be removed

**Comment by Jookia** [comment]
I think this breaks CMake's amount of tests being run.

**Comment by karolherbst** [comment]
yeah, only the gtest test, but this is okay. On error the output gtest output will be printed on travis-ci

**Pull request merged**